bin/passenger-memory-stats

```diff
@@ -107,27 +108,78 @@ class Table
   end
 end
 
+# Parses the specific commandline options.
+#
+# Modeled after `passenger-status` logic, with minor tweaks.
+#
+class CommandLineOptionsParser
+  def parse
+    options = {}
+    parser = create_option_parser(options)
+
+    begin
+      parser.parse!
+      options
+    rescue OptionParser::ParseError => e
+      STDERR.puts e
+      STDERR.puts
+      STDERR.puts "Please see '--help' for valid options."
+
+      exit 1
+    end
+  end
+
+  private
+
+  def create_option_parser(options)
+    OptionParser.new do |opts|
+      opts.banner = "Usage: #{File.basename(__FILE__)} [-h|--help] [--no-apache] [--no-nginx]"
+
+      opts.separator ""
+      opts.separator "Tool for inspecting the application server and Phusion Passenger's memory statistics."
+      opts.separator ""
+
+      opts.separator "Options:"
+
+      opts.on("--no-apache", "Do not display the Apache statistics.") do
+        options[:no_apache] = true
+      end
+      opts.on("--no-nginx", "Do not display the Nginx statistics.") do
+        options[:no_nginx] = true
+      end
+    end
+  end
+end
+
 class App
   def initialize
     @stats = AdminTools::MemoryStats.new
     @colors = Utils::AnsiColors.new
   end
 
-  def start
+  def start(options = {})
```

Comment:
I don't know what the target Ruby version, so I didn't use named parameters, here.